### ISSUE TITLE
Improve readability of TimeSeriesTable when displayed in console

### ISSUE THREAD

**Issue by tkuchida** [opensim-org/opensim-core#1306]
It would be convenient if there were a static method on ConsoleReporter that would allow something like
``` cpp
const int dependentColsToDisplay[3] = {1, 2, 5};
ConsoleReporter::displayTable(myTimeSeriesTable, dependentColsToDisplay);
```
to take advantage of the ConsoleReporter's output formatting, which is easier to read than
``` cpp
std::cout << myTimeSeriesTable << std::endl;
```

**Comment by klshrinidhi** [comment]
I can improve the function that prints `TimeSeriesTable` so that:
```
std::cout << myTimeSeriesTable << std::endl
```
gives more readable result. 
* It looks like one has to go through an indirect method of feeding `TimeSeriesTable` to `ConsoleReporter` in order to see the table in a easily readable format.
* `ConsoleReporter` is not meant to know all the details (metadata) about `TimeSeriesTable`. I think it is better to have this functionality improved in `TimeSeriesTable.h`.

**Comment by tkuchida** [comment]
Ah, ok. I thought it might be cleanest to use the same formatting infrastructure for printing any table of data to the console, which could be shared by ConsoleReporter and TimeSeriesTable—but yes, I think a more readable version of `std::cout << myTimeSeriesTable;` would be an improvement regardless of where the code resides. Thanks!

**Comment by aymanhab** [comment]
I'd add that a method eg toString () would be needed for languages not supporting << operators.

**Comment by klshrinidhi** [comment]
Addressed by #1348.
